File: fun_tau_array.m
function m = fun_tau_array(N, pts, optm)
%% Function to generate time array for Allan variance
% optm = "optimized"    --> logspace array
% optm = "all"          --> complete array
    
    % Most AVAR algorithms do not allow the last m value to be floor(N/2).
    % Apparently, the last m must be smaller than N/2 - length(m)
    max_m = 2^floor(log2(N/2));     % Max m in power of 2
    
    if optm == "all" 
        max_m = N/2 - 1;
        m = 2:max_m;
    elseif optm == "optimized" 
        m = logspace(0, log10(max_m), pts)';
    end
    m = floor(m);        
    m = unique(m);
    
end

File: julia/fun_tau_array.jl
function fun_tau_array(N, pts)
    # Function to generate time array for Allan variance
    # Input: 
        # N     → length of input array
        # pts   → number of correlation points 
    # Output:
        # m     → array of Allan variance correlation time

    # Check input size
    if pts > N/2
        error("Number of points must be less or equal N/2")
    end

    # Calculating correlation time array
    max_m = 2^floor(log2(N/2)); 
    m = 10 .^(range(0, stop=log10(max_m), length=pts));
    m = floor.(Int,m);     
    m = unique(m);

    (m)
end